### PR TITLE
Domain Suggestions TableView Cleanup

### PR DESCRIPTION
**Pull request by frosty**
This PR does a bit of cleanup to the domain suggestions tableview.

First, it removes `RegisterDomainSuggestionsTableViewController`, which just customized a couple of properties, and just uses `DomainSuggestionsTableViewController` directly instead. That VC was originally written to be subclassed and customizable, but in practice it's only used in one place. I also moved the table view controller from the NUX section of the project to the domains section.

This PR also removes some of the customizable properties and changes the defaults of the others to match what was happening in `RegisterDomainSuggestionsTableViewController`. There should be no change in functionality.

**To test**

* Build and run and step through the domain suggestion / selection process, and ensure everything is working as you'd expect.

## Regression Notes

1. Potential unintended areas of impact

I don't think there should be any. Changes are confined to the domain suggestion screen, and should match what we already had.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/DomainSuggestionsTableViewController.swift [hunk 1]
```diff
@@ -8,28 +8,26 @@ protocol DomainSuggestionsTableViewControllerDelegate {
...
 class DomainSuggestionsTableViewController: UITableViewController {
 
     // MARK: - Properties
 
-    open var siteName: String?
-    open var delegate: DomainSuggestionsTableViewControllerDelegate?
-    open var domainSuggestionType: DomainsServiceRemote.DomainSuggestionType = .onlyWordPressDotCom
+    var siteName: String?
+    var delegate: DomainSuggestionsTableViewControllerDelegate?
+    var domainSuggestionType: DomainsServiceRemote.DomainSuggestionType = .noWordpressDotCom
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/DomainSuggestionsTableViewController.swift:20]: 👍

**Review by Gio2018 — APPROVED** [review]
:shipit:

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/DomainSuggestionsTableViewController.swift [hunk 2]
```diff
@@ -8,28 +8,26 @@ protocol DomainSuggestionsTableViewControllerDelegate {
...
 
-    open var useFadedColorForParentDomains: Bool {
-        return true
+    var useFadedColorForParentDomains: Bool {
+        return false
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/DomainSuggestionsTableViewController.swift:23]: Not really necessary to change this, but just wanted to bring up the `return` statement can be omitted here.

**Review by diegoreymendez — APPROVED** [review]
Works as expected.

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/DomainSuggestionsTableViewController.swift (hunk 1 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/DomainSuggestionsTableViewController.swift:20]: I updated this to match the overridden suggestion type that was in the existing DomainSuggestionsTableViewController

**Pull request merged**